lib/mumble-ruby/audio_player.rb

```diff
@@ -64,9 +64,14 @@ def change_volume(pcm_data)
     end
 
     def bounded_produce
+      frame_count = 0
+      start_time = Time.now.to_f
       @file.each_buffer(@encoder.frame_size) do |buffer|
         encode_sample buffer.samples.pack('s*')
         consume
+        frame_count += 1
+        wait_time = start_time - Time.now.to_f + frame_count * 0.01
```

Comment:
I would calculate the necessary wait time based on the sample rate and frame size.
```suggestion
    frame_time = @encoder.frame_size / @sample_rate # Sample Rate needs to also be set in constructor!

        wait_time = start_time - Time.now.to_f + frame_count * frame_time
```

Reply:
Suggestion doesn't work, the wait_time goes negative after few iterations. this is what I got while testing with a wav file :

frame size : 480
 frame time : 0.01
 frame count : 1
wait time1: 0.007859954833984375
frame size : 480
 frame time : 0.01
 frame count : 2
wait time1: 0.007204313278198243
frame size : 480
 frame time : 0.01
 frame count : 3
wait time1: 0.007313041687011718
frame size : 480
 frame time : 0.01
 frame count : 4
wait time1: 0.0068891429901123055
frame size : 480
 frame time : 0.01
 frame count : 5
wait time1: 0.006615447998046878
frame size : 480
 frame time : 0.01
 frame count : 6
wait time1: 0.006233510971069334
frame size : 480
 frame time : 0.01
 frame count : 7
wait time1: 0.005341596603393561
frame size : 480
 frame time : 0.01
 frame count : 8
wait time1: 0.005146102905273439
frame size : 480
 frame time : 0.01
 frame count : 9
wait time1: 0.004216041564941403
frame size : 480
 frame time : 0.01
 frame count : 10
wait time1: 0.0038496017456054743
frame size : 480
 frame time : 0.01
 frame count : 11
wait time1: 0.003392562866210938
frame size : 480
 frame time : 0.01
 frame count : 12
wait time1: 0.0031188678741455034
frame size : 480
 frame time : 0.01
 frame count : 13
wait time1: 0.0023556995391845748
frame size : 480
 frame time : 0.01
 frame count : 14
wait time1: 0.0019973659515380993
frame size : 480
 frame time : 0.01
 frame count : 15
wait time1: 0.001522922515869135
frame size : 480
 frame time : 0.01
 frame count : 16
wait time1: 0.0008505916595459018
frame size : 480
 frame time : 0.01
 frame count : 17
wait time1: 0.0005976390838623169
frame size : 480
 frame time : 0.01
 frame count : 18
wait time1: 0.00019305229187011053
frame size : 480
 frame time : 0.01
 frame count : 19
wait time1: -0.0007916069030761697
frame size : 480
 frame time : 0.01
 frame count : 20
wait time1: -0.001099395751953114

the original code is somehow able to play the audio, however that too was not good.